juju/juju
# Go implementation of Bridge script

## Description of change
add-juju-bridge.py script logic moved to Go code. An example 'juju-bridge' command is provided. 
The code supports multiple statements for interface, inet and inet6 addresses, it does not yet support interface names in pre/post scripts.
Bridge name generation is now done in one place - network/containerizer/bridgepolicy.go:BridgeNameForDevice . It supports long device names (uses a hash to shorten it).  

## QA steps
Run go tests on network, network/debinterfaces and network/containerizer packages.
Create a container machine on a machine with long (15 chars) interface name and check that the bridge was created properly.

## Documentation changes
This change does not change existing behaviour (if a bridge was properly created it will have the same name as before).

## Bug reference
https://bugs.launchpad.net/juju/+bug/1672327
